splink/cluster_metrics.py

```diff
@@ -79,6 +92,139 @@ def _node_degree_sql(
     return sqls
 
 
+def _node_mapping_table_sql(
+    df_node_metrics: SplinkDataFrame,
+):
+    """
+    Generate SQL to make a table that maps composite_unique_ids to 1-indexed integers.
+
+    This is the node-labelling-scheme required by igraph
+    """
+    nodes_table_name = df_node_metrics.physical_name
+    sql_infos = []
+    sql = f"""
+        SELECT
+            composite_unique_id,
+            row_number() OVER(ORDER BY 1) AS rn
+        FROM
+            {nodes_table_name}
+    """
+    row_number_table_name = "__splink__nodes_numbered"
+    sql_info = {"sql": sql, "output_table_name": row_number_table_name}
+    sql_infos.append(sql_info)
+    sql = f"""
+        SELECT
+            composite_unique_id,
+            rn - 1 AS new_id
+        FROM
+            {row_number_table_name}
+    """
+    node_mapping_table_name = "__splink__nodes_integer_mapping"
```

Comment:
might be missing something, but could you not do this as a single statement like `row_number() OVER(ORDER BY 1) - 1 AS new_id`

perhaps that's not supported by some dialects, or I'm misinterpreting the operation?

Reply:
I believe there were difficulties with these kind of 'additional operations' as `row_number` is a window function rather than an ordinary scalar function, but I will double-check

Reply:
This is fine actually - must've been making some other mistake or have messed up the syntax